cli/src/config/templates.toml

```diff
@@ -16,10 +16,10 @@ commit_summary = 'format_commit_summary_with_refs(self, bookmarks)'
 
 annotate_commit_summary = '''
 separate(" ",
-  format_short_id(change_id),
-  format_short_id(commit_id),
-  format_short_signature(author),
-  format_timestamp(committer.timestamp()),
+  change_id.shortest(8),
```

Comment:
This will not show whether the commit is hidden or divergent. 

I'm not sure what to do about it. The simplest things would be to change the color, but it would make the template complicated. Adding `?` for divergent makes sense, but we don't have a short syntax for hidden. 

So, this might not be actionable, but I thought I'd put it out there in case people have ideas.

Reply:
Is the goal of `annotate` to give all the gory details, or to show the last change that modified the line?

Reply:
If the revision is hidden or divergent, it can't be referred to easily by its change id. This seems worth indicating somehow when you show this (potentially useless) change id.

It might be OK to keep things as they are if we also show the commit id. Maybe I'd add color to the change id at some point.

BTW, this keeps reminding me that we might want a syntax like `zzx^a` to refer to a revision with change id zzx (that could be divergent) and commit id starting with `a`. I'm not sure how appropriate this is for hidden commits, but if it is *and* if we decided to usually omit commit ids, it could be a way out. But again, at least for now, I think just showing the commit id might be easier and more convenient (though I'm not certain about it).

Reply:
I feel like it's going to be very rare to want to annotate a hidden or divergent file. Probably rare enough that the user can override the template when they want to do that. I can't remember ever running `hg blame` or `git blame` on a hidden commit.